beamer/tests/contracts/test_request_manager.py

```diff
@@ -121,28 +123,72 @@ def test_claim_counter_challenge(request_manager, token, claim_stake):
     # Do a proper challenge
     request_manager.challengeClaim(claim_id, {"from": challenger, "value": claim_stake + 1})
 
-    # Another party must not be able to join the challenge game
+    # Only the claimer is eligible to outbid the challengers
     with brownie.reverts("Not eligible to outbid"):
         request_manager.challengeClaim(claim_id, {"from": requester})
 
     # The sender of the last challenge must not be able to challenge again
     with brownie.reverts("Not eligible to outbid"):
         request_manager.challengeClaim(claim_id, {"from": challenger})
 
-    # The other party must be able to re-challenge
+    # The other party, in this case the claimer, must be able to re-challenge
     with brownie.reverts("Not enough stake provided"):
         request_manager.challengeClaim(claim_id, {"from": claimer, "value": claim_stake})
     outbid = request_manager.challengeClaim(claim_id, {"from": claimer, "value": claim_stake + 1})
     assert "ClaimMade" in outbid.events
 
-    # Check that the roles are reversed now
-    with brownie.reverts("Not eligible to outbid"):
+    # Check that claimer is leading and cannot challenge own claim
+    with brownie.reverts("Cannot challenge own claim"):
         request_manager.challengeClaim(claim_id, {"from": claimer, "value": 1})
 
-    # The other party must be able to re-challenge, but must increase the stake
+    # The first challenger must be able to re-challenge, but must increase the stake
     with brownie.reverts("Not enough stake provided"):
         request_manager.challengeClaim(claim_id, {"from": challenger, "value": claim_stake})
-    request_manager.challengeClaim(claim_id, {"from": challenger, "value": claim_stake + 1})
+    outbid = request_manager.challengeClaim(
+        claim_id, {"from": challenger, "value": claim_stake + 1}
+    )
+    assert "ClaimMade" in outbid.events
+
+    # Claimer outbids again
+    outbid = request_manager.challengeClaim(claim_id, {"from": claimer, "value": claim_stake + 1})
+    assert "ClaimMade" in outbid.events
+
+    # The second challenger must be able to re-challenge, but must increase the stake
```

Comment:
Why is this the "second" challenger?  Above you say "first challenger", but you're using the same `challenger` address as here.

Reply:
Ah, you're referring to the role, not the address... ignore my comment.

Reply:
Oh I think the comment is wrong. I created its own test for two challengers. This one only tests counter-challenge